openerp/models.py

```diff
@@ -4249,16 +4251,19 @@ def _create(self, cr, user, vals, context=None):
         # check Python constraints
         recs._validate_fields(vals)
 
-        if recs.env.recompute and context.get('recompute', True):
-            result += self._store_get_values(cr, user, [id_new],
+        result += self._store_get_values(cr, user, [id_new],
                 list(set(vals.keys() + self._inherits.values())),
                 context)
+        recs.env.extend_recompute_list(result)
+
+        if recs.env.recompute and context.get('recompute', True):
             result.sort()
             done = []
-            for order, model_name, ids, fields2 in result:
+            for order, model_name, ids, fields2 in recs.env.get_recompute_list():
                 if not (model_name, ids, fields2) in done:
                     self.pool[model_name]._store_set_values(cr, user, ids, fields2, context)
                     done.append((model_name, ids, fields2))
+            recs.env.clear_recompute_list()
```

Comment:
While I understand 
```
class Environment:

    def extend_recompute_old(self, items):
        self.all.recompute_old += items

    def get_recompute_old(self):
        result = self.all.recompute_old
        self.all.recompute_old = []
        return result
```
is cleaner, I am wondering why `_store_set_values` would trigger more recomputation ? Only `get` calls are performed, and the value are stored directly in SQL: The simple call to `get` would lead to field recomputation ?